File: libs/contracts/Dto/DepartmentDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HSMS.contracts.Dto
{
    public class DepartmentDto
    {
        public int Id { get; set; }
        public int BranchId { get; set; }
        public string? BranchName { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Code { get; set; } = string.Empty;
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset UpdatedAt { get; set; }
    }
}

File: apps/HSMS-fe/src/app/models/Department.dto.ts
export interface DepartmentDto {
 id: number;
  branchId: number;
  branchName: string;
  name: string;
  code: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateDepartmentDto {
   branchId: number; 
  name: string;
  code: string;
}